src/evaluators/interpolation/PHAL_NodesToCellInterpolation_Def.hpp

```diff
@@ -70,38 +100,57 @@ void NodesToCellInterpolationBase<EvalT, Traits, ScalarT>::evaluateFields (typen
 
   for (int cell=0; cell<workset.numCells; ++cell)
   {
-    meas = 0.0;
-    for (int qp(0); qp<numQPs; ++qp)
-    {
-      meas += w_measure(cell,qp);
-    }
+    if(interpolationType == CellAverage) {
+      meas = 0.0;
+      for (int qp(0); qp<numQPs; ++qp)
```

Comment:
Not for this PR, but in the future we should avoid looping over qps for affine geometries (since the jacobian is constant). I think you mentioned that Intrepid2 is implementing (or already has) some logic to handle this case. Perhaps we should ask Intrepid2 to do this kind of geometric calculations...

Reply:
I agree, at some point we should use Intrepid2 in a more systematic way. That Intrepid2 optimization should be ready by the end of this FY